pkg/database/token.go

```diff
@@ -397,14 +397,31 @@ func (db *Database) updateStatsTokenInvalid(t time.Time, authApp *AuthorizedApp)
 func (db *Database) updateStatsTokenClaimed(t time.Time, authApp *AuthorizedApp) {
 	t = timeutils.UTCMidnight(t)
 
-	realmSQL := `
+	if err := db.db.Transaction(func(tx *gorm.DB) error {
+		var existing RealmStat
+		if err := tx.
```

Comment:
I'm not sure I see how this is different from the existing implementation?

Reply:
the existing implementation does a blind-write. I wanted to read the data so I can manipulate the existing values. Arrays are hard to increment an index in pure SQL and I have this average calc too.